### ISSUE TITLE
client self signup email addresses are never confirmed

### ISSUE THREAD

**Issue by kkplein** [projectsend/projectsend#540]
Hi,

It seems that when clients self-register, they can use whatever non-existant email address they want, since the address is never actually verified? (could be done using a simple confirmation link that PS sends, and that has to be clicked)

And, as the client login is functional immediately, 'unknown' (unverified) clients can download files that were supposed to be for authenticated users only.

**Comment by mschop** [comment]
Hi,

seems to make sense to me to add a verification. As a workaround until this implemented, you could uncheck the option "Options > Clients > Auto approve new accounts"

**Comment by kkplein** [comment]
Yes, we have done that already. But in fact I think this should not be "Low Priority", as the logon details can be completely bogus, the way it is now. So you think you have a valid way to contact a client, but in fact the only thing that you know for sure, is the ip address. The rest van be completely garbage.
I think this should be *high* prio, as everybody also would expect an email address to be *real* and *functional*, I guess.

**Comment by mschop** [comment]
The only case where this is relevant is, when a customer unintentionally mistyped his own email address. So this feature would only prevent, that you cannot contact the user. Protecting your downloads based on the validity of an email address does not protect your downloads, as a fake email account can be created in few seconds.

I cannot see, where this feature request is crucial for a production site. Sure, this could be implemented to be more prone to mistyped emails but there is no further advantage.

**Comment by kkplein** [comment]
Well, think of the case where you want to allow downloads only for certain known domains (like *@uni.edu) who are manually allowed by the group owner.
If you cannot trust the email address is real and verified, anyone can request access, and provide such a (non existing) email address, and obtain the files.
If a user@uni.edu is **verified**, surely you know that this is a real person studying at uni.edu
Of I know one can create a hotmail in seconds, but hotmail/gmail addresses is not what we're thinking of.

**Comment by ignacionelson** [comment]
I think this falls in the list of privacy items to improve.
Confirming email addresses, even though then can be made with those 10 minute services are good for cases like @kkplein 's where limiting by domain is important.
This, plus asking new users to set a password before the first log in and 2 factor authentication are proposed for the next milestone

**Comment by kkplein** [comment]
Thank you @ignacionelson, good to hear this!

**Comment by mschop** [comment]
Maybe one could additionally add a email-domain whitelist functionality for such use cases. This would prevent from having to manually check the users email address.

**Comment by kkplein** [comment]
That last comment is cool yes. So it would allow sharing files to all registered and verified email addresses from our university plus it's partners. No admin intervention required. Really nice idea @mschop!

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
